### DISCUSSION TITLE
Environment in mkosi.configure script

### DISCUSSION THREAD

**Discussion by ragazenta** [systemd/mkosi#2893] [Q&A]
Hello, I haven't followed `mkosi` for a while and it seems there are some big changes.

Previously I set up `mkosi.configure` script to do some `sed` replacement with some environment variables. I defined the variables in `mkosi.env`. With `mkosi` from latest git main, it doesn't work anymore, the variables are now blank.

Is there any alternatives to do this?

Thanks.

**Comment by NekkoDroid** [comment]
Is the `mkosi.configure` script in one of the subimages and the `mkosi.env` in the top level? If so you need to use `PassEnvironment=` to list the variables that should be passed to subimages (assuming you are using the latest git main)

**Reply by ragazenta** [reply]
Yes, I've tried `PassEnvironment=` but didn't work. So I moved the `mkosi.env` to subimage directory.
